hypersync-format/src/types/bloom_filter_wrapper.rs

```diff
@@ -25,6 +26,29 @@ impl FilterWrapper {
     pub fn insert_hash(&mut self, hash: u64) -> bool {
         self.0.insert_hash(hash)
     }
+
+    /// Creates a bloom filter out of a vec of Addresses.  Suitable for using
+    /// as a query param that takes a filter.
+    /// bits_per_key defaults to 16
+    pub fn from_addresses(addresses: Vec<Address>, bits_per_key: Option<usize>) -> Self {
+        let bits_per_key = if let Some(bits) = bits_per_key {
+            bits
+        } else {
+            16
+        };
+
+        let mut filter = FilterWrapper::new(bits_per_key, addresses.len());
+
+        // first put into hash set to remove duplicates
+        let addresses: HashSet<Address> = addresses.into_iter().collect();
```

Comment:
Can collect into IntMap<u64> directly it is more efficient but not critical

Reply:
Using intset :+1: